### PR TITLE
Add delete organization confirmation input field

### LINKED ISSUE #765: Bug Report: Change "Delete this Organization" button to red.
**Describe the bug**
The current org dashboard has a "Delete organization" button that is colored green, which can be confusing for users and potentially lead to accidental deletions. To improve the user experience and avoid accidental deletions, I propose changing the color of the "Delete this organization" button from green to red.

A red "Delete organization" button will help to differentiate it from other buttons and actions within the system, making it more noticeable and helping users to take extra care before taking any irreversible actions. The color red is often associated with danger or caution, which is appropriate for this action as it permanently deletes the entire organization from the system. The new color will also serve as a visual cue to indicate the potential risk associated with the action and prompt users to double-check before proceeding. This also includes switching the colors of the "yes/no" buttons in the confirmation modal.

**Screenshots**

issues

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Kelvyn2012**
**What kind of change does this PR introduce?**

A feature
Change the delete button to Red.
Created an input field to confirm the organization name before deleting.

dashboard delete button
dashboard delete confirmation

Fixes #765

**Did you add tests for your changes?**

 No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

File: public/locales/en.json [hunk 1]
```diff
@@ -109,6 +109,7 @@
     "membershipRequests": "Membership Requests",
     "deleteOrganization": "Delete Organization",
     "deleteMsg": "Do you want to delete this organization?",
+    "confirmDeleteText": "Please type <bold>{{orgName}}</bold> to confirm",
```
Comment (palisadoes) [inline public/locales/en.json:112]: You need to make this work in all languages. Use Google Translate for the others.

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. Support all languages.
2. Where in the code are you verifying that the person has typed the organization name correctly before doing the deletion?  What happens if they type in the wrong name?

**Comment by kayceeDev** [comment]
In addition to the @palisadoes comments, You can improve this by greying out the YES button till a user enters an organization name that matches the organization he drilled down on.

To implement this, you can check on each user input. You can use debounce input to make it seamless and avoid lags.
So you have to save the organization name to state once a user drills down to a particular organization if it is not there already.

Also, there is a PR open to moving the "Delete organization" button inside settings. So if this PR gets merged, you may have to revisit it once the other PR gets merged too. But in all, you did a great job coming up with this UX.

File: public/locales/en.json (hunk 1 above)
Comment (kayceeDev) [reply public/locales/en.json:112]: > Regarding your second question, the YES button remains disabled as long as the user types a wrong organization name in the input field or leaves the input field empty, the YES button is only enabled when the user get the organization name correctly and its also case sensitive.

I mean style the button and make the color faint on disabled mode. Maybe make it less green.

**Comment by Kelvyn2012** [comment]
Hi @kayceeDev, thanks for the review. This has been implemented already. They `Yes` button is greyed out until the user enters the matching organization name & this is checked on each user input. Please let me know if I missed something here.

> In addition to the @palisadoes comments, You can improve this by greying out the YES button till a user enters an organization name that matches the organization he drilled down on.
> 
> To implement this, you can check on each user input.



I'll do some research about the debounce input and get back to you. I don't think we need to save the organization name to a state as it is already in the component and will not be updated. 

> You can use debounce input to make it seamless and avoid lags.
> So you have to save the organization name to state once a user drills down to a particular organization if it is not there already.

**Comment by kayceeDev** [comment]
> Hi @kayceeDev, thanks for the review. This has been implemented already. They `Yes` button is greyed out until the user enters the matching organization name & this is checked on each user input. Please let me know if I missed something here.
> 
> > In addition to the @palisadoes comments, You can improve this by greying out the YES button till a user enters an organization name that matches the organization he drilled down on.
> > To implement this, you can check on each user input.
> 
> I'll do some research about the debounce input and get back to you. I don't think we need to save the organization name to a state as it is already in the component and will not be updated.
> 
> > You can use debounce input to make it seamless and avoid lags.
> > So you have to save the organization name to state once a user drills down to a particular organization if it is not there already.
> 
> Thank you!!

Nice one. Take a look at the other input in the codebase. I think the debounce input was used.

File: public/locales/en.json (hunk 1 above)
Comment (Kelvyn2012) [reply public/locales/en.json:112]: @palisadoes i have made the required changes, added language support for the languages and also made use of debounce input.

**Review by palisadoes — APPROVED** [review]

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. The deletion of an organization is a serious matter. 
2. You will need to write tests for this code to make sure it works properly

**Comment by palisadoes** [comment]
Please fix the failing tests. Our test code coverage system will fail if either of these two conditions occur:

1) The overall code coverage drops below the target threshold of the repository
2) Any file in the pull request has code coverage levels below the repository threshold

We do this to ensure the overall reliability of the code base is maintained.

**Comment by palisadoes** [comment]
Please fix the failing tests. Our test code coverage system will fail if either of these two conditions occur:

1) The overall code coverage drops below the target threshold of the repository
2) Any file in the pull request has code coverage levels below the repository threshold

We do this to ensure the overall reliability of the code base is maintained.

**Comment by TheoCathy** [comment]
@Kelvyn2012 is the "No" btn also grayed out/disabled ? It shouldn't be. For easy cancelling incase the user changes their mind.

**Comment by Kelvyn2012** [comment]
> @Kelvyn2012 is the "No" btn also grayed out/disabled ? It shouldn't be. For easy cancelling incase the user changes their mind.
@TheoCathy no it is not disabled or grayed out, just the 'YES' button is disabled till you confirmed by typing the name of the organization in the input

**Comment by palisadoes** [comment]
Please fix the failing tests. Our test code coverage system will fail if either of these two conditions occur:

1) The overall code coverage drops below the target threshold of the repository
2) Any file in the pull request has code coverage levels below the repository threshold

We do this to ensure the overall reliability of the code base is maintained. This is a prerequisite for getting your PR approved. Assigned reviewers regularly review the PR queue and tend to focus on PRs that are passing.

**Comment by Kelvyn2012** [comment]
@palisadoes  please i am still working on the test

File: src/screens/OrganizationDashboard/OrganizationDashboard.test.tsx [hunk 2]
```diff
@@ -101,4 +104,36 @@ describe('Organisation Dashboard Page', () => {
...
+    fireEvent.click(screen.getByText('Delete This Organization'));
+    const orgInput = screen.getByTestId(/orgName/i);
+    userEvent.type(orgInput, 'bar');
+    expect(screen.getByTestId(/deleteOrganizationBtn/i)).toHaveProperty(
+      'disabled',
+      true
+    );
+
+    userEvent.clear(orgInput);
+    userEvent.type(orgInput, 'Dummy Organization');
```
Comment (Kelvyn2012) [inline src/screens/OrganizationDashboard/OrganizationDashboard.test.tsx:132]: @palisadoes @kayceeDev userEvent.type is not triggering the onChange event in the test, but works fine locally.
I would appreciate any assistance with this.

**Comment by palisadoes** [comment]
Please ask the assigned reviewers

**Comment by palisadoes** [comment]
You have six assigned.

File: src/screens/OrganizationDashboard/OrganizationDashboard.test.tsx (hunk 2 above)
Comment (Rumone) [reply src/screens/OrganizationDashboard/OrganizationDashboard.test.tsx:132]: What have you tried?

Comment (Kelvyn2012) [reply src/screens/OrganizationDashboard/OrganizationDashboard.test.tsx:132]: https://github.com/testing-library/user-event/issues/676
I have tried all solutions here @Rumone

Comment (Rumone) [reply src/screens/OrganizationDashboard/OrganizationDashboard.test.tsx:132]: > Adding userEvent.tab() after the userEvent.type() also works. 

You tried this as well.

Comment (Kelvyn2012) [reply src/screens/OrganizationDashboard/OrganizationDashboard.test.tsx:132]: yes i did that also

**Comment by palisadoes** [comment]
Please fix the failing tests

**Comment by palisadoes** [comment]
Closing. It's been two weeks without an update on fixing the failing tests

**Comment by TheoCathy** [comment]
@palisadoes kindly assign this to me

**Pull request closed without merging**